app/src/main/java/de/danoeh/antennapod/fragment/FeedItemlistFragment.java

```diff
@@ -314,13 +315,18 @@ public void onEventMainThread(FeedItemEvent event) {
         if (feed == null || feed.getItems() == null) {
             return;
         }
-        for (int i = 0, size = event.items.size(); i < size; i++) {
-            FeedItem item = event.items.get(i);
-            int pos = FeedItemUtil.indexOfItemWithId(feed.getItems(), item.getId());
-            if (pos >= 0) {
-                feed.getItems().remove(pos);
-                feed.getItems().add(pos, item);
-                adapter.notifyItemChangedCompat(pos);
+
+        if (feed.isLocalFeed()) {
+            LocalFeedUpdater.updateFeed(feed, requireContext().getApplicationContext(), null);
```

Comment:
The fragments should not do any parsing.

> Actual list of items did not want to update for me

There must be another way, for example refreshing it in the background when the file is deleted.